tsdb/internal/fieldsindex.proto

```diff
@@ -32,3 +32,13 @@ message Field {
 message MeasurementFieldSet {
 	repeated MeasurementFields Measurements = 1;
 }
+
+message MeasurementFieldChange {
```

Comment:
Given that our operations are `delete an entire measurement` and `add a field`, it might be cleaner to have something like:

```
message MeasurementFieldChange {
  oneof change_type {
    // alternatively these could be special measurement types
    MeasurementFields add_fields = 1;
    string measurement_to_delete = 2;
  }
}
```

Otherwise the code is somewhat confusing when we handle a MeasurementFieldChange, and it isn't clear that a single field cannot be deleted.

Reply:
After messing around with this this, I'm going to push back on this suggestion. We're still going to have to encapsulate a Delete flag in the FieldChange object, and setting it directly from the protobuf yields much cleaner code than extracting the information from the pseudo-union.  It's sort of a mismatch between Go and protobuf models of the world.

The only way to make this clean is a rewrite of FieldChange into an interface with a DoOperation() which either creates or deletes, depending on the object implementing the FieldChange interface. That seems too complex for such a limited scope of operation.  As always, happy to discuss.

Reply:
At least we should have a comment somewhere describing the currently valid operations, especially that measurement and field filled with delete=true is invalid.

Maybe it would be enough for now to have, instead of `bool Delete`, and `enum ChangeType` of operations? And have `AddMeasurementField` and `DeleteMeasurement` as the possible enum values.